sdv/constraints/tabular.py

```diff
@@ -334,6 +334,12 @@ def _transform(self, table_data):
             pandas.DataFrame:
                 Transformed data.
         """
+        # To make the NaN to None mapping work for pd.Categorical data, we need to convert
+        # the columns to object before replacing NaNs with None.
+        for column in self._columns:
+            if pd.api.types.is_categorical_dtype(table_data[column]):
+                table_data[column] = table_data[column].astype(object)
+
```

Comment:
Adding the np.bytes revealed an actual bug in the FixedCombinations with pd.Categorical and NaNs. It was not caught before due to reject sampling; before enough rows were generated, but none with a combination including NaNs in the categorical column. After adding np.bytes and somehow a bit randomly, the synthesizer was able to generate only 9 out of the 10 rows. Changing the number of rows to sample also made the test pass but did not fix the bug, haha. This fixes the bug, and I added a test for it. Let me know if it makes sense

Reply:
I'm confused why adding bytes revealed this when the check is for is_categorical_dtype. Bytes are not a categorical dtype